pkg/connector/tickets.go

```diff
@@ -0,0 +1,464 @@
+package connector
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"path"
+	"strconv"
+	"time"
+
+	v2 "github.com/conductorone/baton-sdk/pb/c1/connector/v2"
+	"github.com/conductorone/baton-sdk/pkg/annotations"
+	"github.com/conductorone/baton-sdk/pkg/pagination"
+	sdkTicket "github.com/conductorone/baton-sdk/pkg/types/ticket"
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	jira "github.com/conductorone/go-jira/v2/cloud"
+)
+
+type TicketManager interface {
+	GetTicket(ctx context.Context, ticketId string) (*v2.Ticket, annotations.Annotations, error)
+	CreateTicket(ctx context.Context, ticket *v2.Ticket, schema *v2.TicketSchema) (*v2.Ticket, annotations.Annotations, error)
+	GetTicketSchema(ctx context.Context, schemaID string) (*v2.TicketSchema, annotations.Annotations, error)
+	ListTicketSchemas(ctx context.Context, pToken *pagination.Token) ([]*v2.TicketSchema, string, annotations.Annotations, error)
+}
+
+func ticketBuilder(j *Jira) TicketManager {
+	return j
+}
+
+func getJiraStatusesForProject(ctx context.Context, client *jira.Client, projectId string) ([]jira.JiraStatus, error) {
```

Comment:
As it turns out, not providing a project id seemed to only give me global statuses, which didnt include all the statuses... If I include projectId it gives me the correct statuses

you would think 
`rest/api/3/statuses/search?expand=usages&startAt=0&maxResults=100&statusCategory=DONE&projectId=10000`
 returns a subset of statuses from 
`rest/api/3/statuses/search?expand=usages&startAt=0&maxResults=100&statusCategory=DONE`
 but they are different...